build/circle-deps.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+set -ex
+
+if [ "$1" = "docker" ]; then
```

Comment:
Add a comment about how this script is run first with no arguments then re-runs itself inside the container with the arg `docker` (actually, why not make this block a second script?)